pkg/controller/statusmanager/status_manager.go

```diff
@@ -171,6 +155,17 @@ func (status *StatusManager) SetDeployments(deployments []types.NamespacedName)
 	status.deployments = deployments
 }
 
+func (status *StatusManager) SetRelatedObjects(objects []configv1.ObjectReference) {
+	co := &configv1.ClusterOperator{ObjectMeta: metav1.ObjectMeta{Name: status.name}}
+	err := status.client.Get(context.TODO(), types.NamespacedName{Name: status.name}, co)
+	isNotFound := errors.IsNotFound(err)
+	if err != nil && !isNotFound {
+		log.Printf("Failed to get ClusterOperator %q: %v", status.name, err)
+		return
+	}
+	co.Status.RelatedObjects = objects
```

Comment:
This is just modifying the copy that you just fetched, which will be GC'ed as soon as this function returns.

You can just store the objects in a field on `status` here, and then set them on the object in `Set`